common/omv_csi.c

```diff
@@ -540,14 +547,16 @@ int omv_csi_probe(omv_i2c_t *i2c) {
         // Special case for OV5640.
         #if (OMV_OV5640_REV_Y_CHECK == 1)
         if (csi->chip_id == OV5640_ID && HAL_GetREVID() < 0x2003) {
-            csi->clk_hz = OMV_OV5640_REV_Y_FREQ;
+            clk_hz = OMV_OV5640_REV_Y_FREQ;
         }
         #endif
 
-        // Allow reconfiguring (or disabling) the external clock
-        // if just one sensor is detected, or for main sensors.
-        if (dev_count == 1 || !csi->auxiliary) {
-            omv_csi_set_clk_frequency(csi->clk_hz);
+        // Sensors can request disabling the clock (clk_hz == 0). This
+        // is allowed only if the clock is not shared (dev_count == 1).
+        // If the frequency is valid, the clock can be reconfigured as
+        // all sensors should share a common frequency.
+        if (clk_hz != 0 || dev_count == 1) {
+            omv_csi_set_clk_frequency(csi, clk_hz);
```

Comment:
Reviewed the PR. Just the change with omv_csi_set_clk_frequency() here and the removal below stand out.

The FLIR Lepton and PAG7936 are both 24 MHz... and then OV5640 + Lepton are 24 MHz both or 24 and 25 MHz if rev y. 

So, the above kinda works, but, the aux sensor could override the main sensor clock in this case for like the rev y? Why not keep the previous behavior?

Reply:
The assumption is that if you use multiple sensors, they must all either share a common clock frequency, or use separate clock sources (internal clock, onboard OSC, or maybe a different timer). For an OV5640, you could either use an onboard OSC or drop the REV-Y thing.

Reply:
The N6 can't do an exact 24MHz either, doesn't seem to be a problem. Note please review the sensors drivers and see which need exact/nominal frequency.

Reply:
Approved

Reply:
Will remove it. I think I have one of these rev-y boards around too. So, I can test.